src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs

```diff
@@ -274,5 +309,25 @@ private static bool IsApplicableTriggerCharacter(string triggerCharacter, RazorL
             // Unknown trigger character.
             return false;
         }
+
+        private class CompletionComparer : IEqualityComparer<CompletionItem>
+        {
+            public static CompletionComparer Instance = new CompletionComparer();
+
+            public bool Equals(CompletionItem x, CompletionItem y)
+            {
+                return x.InsertText.Equals(y.InsertText, StringComparison.Ordinal)
```

Comment:
You'll want to compare the Label only because that's what will show up. For instance, once C# moves to snippet completions their insert text will be more complex and we wont want to be adding our C# completion items in that case.